tests/integration/test_integration.py

```diff
@@ -70,64 +70,137 @@ async def build_charms_and_deploy_vault(ops_test: OpsTest):
         k8s_namespace = ops_test.model.name
         crash_pod(name=f"{APPLICATION_NAME}-1", namespace=k8s_namespace)
-        await ops_test.model.wait_for_idle(
-            apps=[APPLICATION_NAME],
-            status="active",
-            timeout=1000,
-            wait_for_exact_units=NUM_VAULT_UNITS,
-        )
+        await wait_for_vault_status_message(
+            ops_test, count=1, expected_message="Waiting for vault to be unsealed", timeout=300
+        )
+        unit_addresses = [row.get("address") for row in await read_vault_unit_statuses(ops_test)]
+        unseal_vault(unit_addresses[crashing_pod_index], root_token, unseal_key)
+        async with ops_test.fast_forward(fast_interval="10s"):
+            await ops_test.model.wait_for_idle(
+                apps=[APPLICATION_NAME],
+                status="active",
+                timeout=1000,
+                wait_for_exact_units=NUM_VAULT_UNITS,
+            )
 
     @pytest.mark.abort_on_fail
     async def test_given_application_is_deployed_when_scale_up_then_status_is_active(
-        self, ops_test: OpsTest, build_charms_and_deploy_vault: dict[str, Path | str]
+        self,
+        ops_test: OpsTest,
+        build_charms_and_deploy_vault: dict[str, Path | str],
+        initialize_leader_vault,
     ):
         assert ops_test.model
-        num_units = NUM_VAULT_UNITS + 2
+        _, root_token, unseal_key = initialize_leader_vault
+        num_units = NUM_VAULT_UNITS + 1
         app: Application = ops_test.model.applications[APPLICATION_NAME]
         await app.scale(num_units)
 
-        await ops_test.model.wait_for_idle(
-            apps=[APPLICATION_NAME],
-            status="active",
-            timeout=1000,
-            wait_for_exact_units=num_units,
+        await wait_for_vault_status_message(
+            ops_test, count=1, expected_message="Waiting for vault to be unsealed", timeout=300
         )
+        unit_addresses = [row.get("address") for row in await read_vault_unit_statuses(ops_test)]
+        unseal_vault(unit_addresses[-1], root_token, unseal_key)
+
+        async with ops_test.fast_forward(fast_interval="10s"):
+            await ops_test.model.wait_for_idle(
+                apps=[APPLICATION_NAME],
+                status="active",
+                timeout=1000,
+                wait_for_exact_units=num_units,
+            )
 
     @pytest.mark.abort_on_fail
     async def test_given_application_is_deployed_when_scale_down_then_status_is_active(
-        self, ops_test: OpsTest, build_charms_and_deploy_vault: dict[str, Path | str]
+        self,
+        ops_test: OpsTest,
+        build_charms_and_deploy_vault: dict[str, Path | str],
+        initialize_leader_vault,
     ):
         assert ops_test.model
+        _, root_token, _ = initialize_leader_vault
         app: Application = ops_test.model.applications[APPLICATION_NAME]
-        await app.scale(NUM_VAULT_UNITS)
 
+        # TODO: maybe get the app ip and connect with cert?
```

Comment:
Should we keep this?

Reply:
i'll just remove it this pr is already large enough.